carrot_cli/src/carrot_cli/helper/carrot_helper.py

```diff
@@ -0,0 +1,901 @@
+        test_workflow_outputs = self._get_wdl_outputs(test_wdl_local)
+        if not bool(test_workflow_outputs):
+            # TODO: skip the pipeline instead
+            raise Exception("No supported result type for the pipeline.")
+        test_wdl = self._get_online_path(f"{WDLS_DIR}/{pipeline_name}.wdl")
+        test_wdl_checksum = self.get_checksum(test_wdl_local)
+
+        pipeline = self.pipelines.get(pipeline_name)
+        if not pipeline:
+            pipeline = Pipeline(
+                **self.call_carrot(
+                    "pipeline", "create", field_to_uuid="pipeline_id"))
+
+        for template_name in template_names:
+            if template_name in pipeline.templates and \
+                    (pipeline.templates[template_name].test_wdl_checksum ==
+                     test_wdl_checksum and
+                     pipeline.templates[template_name].eval_wdl_checksum ==
+                     self.get_checksum(self._get_eval_wdl_local_path(
+                         pipeline_name, template_name))):
+                continue
+            template = self.create_template(
+                pipeline_name, pipeline, template_name, test_wdl,
+                test_wdl_checksum, test_workflow_outputs)
+            pipeline.templates[template_name] = template
+
+        return pipeline
+
+    def get_pipelines_from_dirs(self, dirs):
+        """
+        Takes a list of directories as input, traverses every
+        directory and sub-directories and extracts pipelines.
+        """
+        pipelines = {}
+        wd = self.working_dir
+        dirs = [x for x in dirs if os.path.isdir(x)]
+        for d in dirs:
+            d = os.path.normpath(d).split(os.sep)
+            if len(d) == 0:
+                continue
+            pipeline = d[0]
+            templates = [d[1]] if len(d) == 2 else None
+            run_dirs = [d[2]] if len(d) == 3 else None
+            print("\n----", d, "\n", pipeline, "\n")
+            exit()
+            if pipeline not in pipelines:
+                pipelines[pipeline] = {}
+            templates = \
+                templates or \
+                [x for x in next(os.walk(os.path.join(wd, pipeline)))[1]]
+            for t in templates:
+                if t not in pipelines[pipeline]:
+                    pipelines[pipeline][t] = []
+                run_dirs = \
+                    run_dirs or \
+                    [x for x in next(os.walk(os.path.join(
+                        wd, pipeline, t)))[1]]
+                for r in run_dirs:
+                    if r not in pipelines[pipeline][t]:
+                        pipelines[pipeline][t].append(r)
+        return pipelines
+
+    @staticmethod
+    def persist_object(obj, filename):
+        with open(filename, "w") as f:
+            json.dump(
+                obj, f,
+                default=lambda o: o.__dict__,
+                indent="\t",
+                sort_keys=True)
+
+    def call_carrot(
```

Comment:
Places where this is called can probably be replaced with calls into the rest modules, since most of the carrot_cli commands are just calling into those and sometimes doing some brief input processing.